### DISCUSSION TITLE
Draw Routes between 2 points on offline status

### DISCUSSION THREAD

**Discussion by FashionCStar** [rnmapbox/maps#1306] [Q&A] [answered]
I have downloaded offline packs for specific areas.
And after set offline, I want to draw the routes between 2 points in the downloaded pack

I found this document from the mapbox SDK docs
https://docs.mapbox.com/android/navigation/guides/offline-service/

![image](https://user-images.githubusercontent.com/39269334/113689912-ae0c8080-967f-11eb-80ac-2766fad749b9.png)

How could I use this feature in react-native-mapbox-gl?

```
      const res = await directionsClient.getDirections(reqOptions).send();
      this.setState({
        route: makeLineString(res.body.routes[0].geometry.coordinates),
      });
```

directionsClient.getDirections() function works well on online status, but getting error {isTrusted:false} on offline status.

**Comment by ferdicus** [comment, marked as answer]
The Navigation SDK is not included in our project. 

Please see here for the packages we include (and support):
https://github.com/react-native-mapbox-gl/maps/blob/a1a95722b4a7f3da35ef48e258785fc79a02b39a/android/rctmgl/build.gradle#L43-L45

**Reply by FashionCStar** [reply]
```
import MapboxDirectionsFactory from '@mapbox/mapbox-sdk/services/directions';

const clientOptions = {accessToken: 'pk.eyJ1IjoiZmFzaGlvbmRldjEiLCJhIjoiY2tscjlmM3diMTNkaTJvbnc1OXBpbzVwNiJ9.yJIaRGA7FoRXjQCSPj3WEA'};
const directionsClient = MapboxDirectionsFactory(clientOptions);
const res = await directionsClient.getDirections(reqOptions).send();
this.setState({
    route: makeLineString(res.body.routes[0].geometry.coordinates),
});

```
Of course, I am using mapbox-sdk to get directions.
But it doesnt work on offline status
I want to get directions data on offline status
